03/app.js

```diff
@@ -1,21 +1,79 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-
-import Category from './Category';
-import Cart from './Cart';
-
+import Category from './Components/Category';
+import Cart from './Components/Cart';
+import Product from './Components/Product';
 import data from './data.json';
+import { v4 as uuid } from 'uuid';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
-    
+    addToBasket = newProduct => {
+        if (this.state.cart.includes(newProduct)) {
+            return
+        } else {
+            this.setState(state => {
+                return {
+                    cart: [...state.cart, newProduct ]
+                }
+            })
+        }
+
+    }
+    removeFromBasket = id => {
+        const cart = this.state.cart.filter(item => item.id !== id)
+        this.setState({
+            cart
+        })
+    }
+    // ta funkcja zawsze zwraca mi to samo, czyli działa źle. gdzie szukać błędu?
+    getNewKey = () => this.state.cart.reduce((acc, item) => acc > item.id ? acc : item.id, 0) + 1;
```

Comment:
Nie wiem dlaczego chcesz generować nowy key. Masz id dla produktu i to wystarczy.
Może problemem jest to, że przeszukujesz `cart`, który jest pusty dopóki ktoś nie "zakupi" produktu.

Reply:
bo w ten sposób w sklepie internetowym można kupić tylko po jednej sztuce produktu na klienta, jak dodasz np. dwie myszki komputerowe to id będzie się już duplikowało. później dopiero zauważyłem że chcesz tylko jeden produkt na raz.